IGHTMLQuery/Classes/IGXMLNodeSet.h

```diff
@@ -13,16 +13,16 @@
 
 @class IGXMLNode;
 
-typedef void (^IGXMLNodeSetEnumerateBlock)(IGXMLNode* node, NSUInteger idx, BOOL *stop);
+typedef void (^IGXMLNodeSetEnumerateBlock)(IGXMLNode* _Nonnull node, NSUInteger idx, BOOL* _Nullable stop);
 
-typedef void (^IGXMLNodeSetEachBlock)(IGXMLNode* node);
+typedef void (^IGXMLNodeSetEachBlock)(IGXMLNode* _Nonnull node);
 
 @protocol IGXMLNodeSetCore <IGHTMLQueryJavaScriptExport>
 
 /**
  nodes in this node set
  */
-@property (nonatomic, copy, readonly) NSOrderedSet* nodes;
+@property (nullable, nonatomic, copy, readonly) NSOrderedSet* nodes;
```

Comment:
I can agree with that.

Note tho that there are other places in the `IGXMLNodeSet` implementation that then could be simplified. If `_nodes` can NEVER be `nil` then there's no need to have if-else expressesions in a few places (e.g. the `-allObjects` implementation). If it's agreed upon that `nodes` (and `_nodes`) can never be nil, I can make appropriate changes. Just let me know.